eo-runtime/src/main/java/EOorg/EOeolang/EOcage.java

```diff
@@ -108,4 +112,96 @@ public Phi lambda() {
             return new Data.ToPhi(true);
         }
     }
+
+    /**
+     * Class to trace if the cage got into recursion during the dataization.
+     * @since 0.36
+     * @todo #2836:60min Add a new parameter of recursion depth. This parameter
+     *   should be set by user via pom.xml. We can make DATAIZING_CAGES a
+     *   ConsurrentHashMap which will add thread safety to the class.
```

Comment:
@levBagryansky Maybe it's better to split this `todo` into two separate puzzles:
1) The first one for "recursion depth"
2) The second one for "thread-safety"

Reply:
@volodya-lombrozo done